framework/src/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -221,19 +225,96 @@ public int hashCode() {
         return result;
     }
 
-    /** Returns the annotation mirror as specified by the "qualifier" value in metaAnno. */
-    private AnnotationMirror getAnnotationMirrorOfQualifier(AnnotationMirror metaAnno) {
+    /**
+     * Returns the annotation mirror as specified by the "qualifier" value in {@code qualifierAnno}.
+     * If {@code argumentAnno} is specified, then arguments are copied from {@code argumentAnno} to
+     * the returned annotation, renamed according to {@code argumentMap}.
+     *
+     * @param qualifierAnno annotation specifying the qualifier class
+     * @param argumentAnno annotation containing the argument values
+     * @param argumentMap map of argument names, which maps from names in {@code argumentAnno} to
+     *     names used in the returned annotation
+     */
+    private AnnotationMirror getAnnotationMirrorOfQualifier(
+            AnnotationMirror qualifierAnno,
+            AnnotationMirror argumentAnno,
+            Map<String, String> argumentMap) {
```

Comment:
I suggest naming variables not according to their type (such as `argumentMap`), but according to their semantics or their use.  For example, this could be called `argumentRenaming` or something similar.